scipy/special/wright_bessel.pxd

```diff
@@ -0,0 +1,685 @@
+# -*-cython-*-
+#
+# Implementation of Wright's generalized Bessel function Phi [1].
+#
+# Copyright: Christian Lorentzen
+#
+# Distributed under the same license as SciPy
+#
+#
+# Implementation Overview:
+#
+# So far, only non-negative values of rho=a, beta=b and z=x are implemented.
+# There are 5 different approaches depending on the ranges of the arguments.
+# 1. Taylor series expansion in x=0 [1], for x <= 1.
+#    Involves gamma funtions in each term.
+# 2. Taylor series expansion in x=0 [2], for large a.
+# 3. Taylor series in a=0, for tiny a and not too large x.
+# 4. Asymptotic expansion for large x [3, 4].
+#    Suitable for large x while still small a and b.
+# 5. Integral representation [5], in principle for all arguments.
+#
+# References:
+# [1] https://dlmf.nist.gov/10.46.E1
+# [2] P. K. Dunn, G. K. Smyth (2005), Series evaluation of Tweedie exponential
+#     dispersion model densities. Statistics and Computing 15 (2005): 267-280.
+# [3] E. M. Wright (1935), The asymptotic expansion of the generalized Bessel
+#     function. Proc. London Math. Soc. (2) 38, pp. 257-270.
+#     https://doi.org/10.1112/plms/s2-38.1.257
+# [4] R. B. Paris (2017), The asymptotics of the generalised Bessel function,
+#     Mathematica Aeterna, Vol. 7, 2017, no. 4, 381 - 406,
+#     https://arxiv.org/abs/1711.03006
+# [5] Y. F. Luchko (2008), Algorithms for Evaluation of the Wright Function for
+#     the Real Arguments' Values, Fractional Calculus and Applied Analysis 11(1)
+#     http://sci-gems.math.bas.bg/jspui/bitstream/10525/1298/1/fcaa-vol11-num1-2008-57p-75p.pdf
+
+import cython
+from libc.math cimport (cos, exp, floor, fmax, fmin, log, log10, pow, sin,
+                        sqrt, M_PI)
+
+cdef extern from "_c99compat.h":
+    int sc_isnan(double x) nogil
+    int sc_isinf(double x) nogil
+
+from ._cephes cimport Gamma, lgam, rgamma, zeta, sinpi, cospi
+from ._digamma cimport digamma
+from ._complexstuff cimport inf, nan
+from . cimport sf_error
+
+# Euler Gamma constant aka Euler Mascheroni constant
+DEF M_EG = 0.57721566490153286060
+# zeta(3)
+DEF M_Z3 = 1.2020569031595942854
+# rgamma_zero: smallest value x for which rgamma(x) == 0 as x gets large
+DEF rgamma_zero = 178.47241115886637
+# exp_inf: smallest value x for which exp(x) == inf
+DEF exp_inf = 709.78271289338403
+# exp_zero: largest value x for which exp(x) == 0
+# DEF exp_zero = -745.13321910194117
+
+
+# 1. Taylor series expansion in x=0, for x <= 1
+#    Phi(a, b, x) = sum_k x^k / k! / Gamma(a*k+b)
+#
+# Note that every term, and therefore also Phi(a, b, x), is monotone decreasing
+# with increasing a or b.
+@cython.cdivision(True)
+cdef inline double _wb_series(double a, double b, double x,
+    unsigned int nstart, unsigned int nstop) nogil:
+    cdef:
+        unsigned int k, k_max
+        double xk_k, res
+
+    xk_k = x**nstart * Gamma(nstart + 1)  # x^k/k!
+    res = xk_k * rgamma(nstart*a + b)
```

Comment:
Computing a ratio of gammas in this way tends to lead to avoidable overflow-you usually want to do something like use the Lanczos approximation (it's internally available in `special`) to cancel the large factors analytically.

Reply:
(You will also see various places using `exp` + `gammaln` for this purpose, but that is strictly less accurate since you can get cancellation when subtracting the gamma terms.)

Reply:
Good hint with the Lanczos approx!
In this case, there is/was a typo, it should read
```python
xk_k = x**nstart * rgamma(nstart + 1)
```
Then it is clear that is is not the ratio of Gammas but a multiplication. Each term  is just `x^k / (k! * Gamma(k*a+b))`.